wordpress-mobile/WordPress-iOS
# MediaLibrary: MediaImageExporter and tests.

Continuing from #7117, this PR introduces the `MediaImageExporter` which does a few good bits of work.

Firstly, exporter errors are propagated upwards via `throw` as custom `enum` values so we can track exactly what went wrong in the process. The localized strings that result, however, should be generally written for a user to read. System errors are instead propagated as is via `MediaExportSystemError`.

The `ImageSourceWriter` utilizes CoreGraphics to do the heavy lifting of actually resizing images and writing them to disk. The comments _should_ help explain what's going on.

The goal was to consolidate how and where we process images for Media from a few places to just one. For example, the upcoming `MediaURLExporter` utilizes the `MediaImageExporter` to process images it detects as a URL, versus trying to do it itself. We'll also generate thumbnails of existing media via the same code in an upcoming implementation.

While there's a lot going here, the unit tests coverage should give us a nice handle on the implementation and ensuring the images are being processed as expected. For example, we've had long-running support issues with images not being resized as the user expects, based on the settings in app settings.

To test:
1. Make sure the code compiles.
2. Run the tests for `MediaImageExporterTests.swift`.

Needs review: @frosty and @SergioEstevao round three :surfer:
